src/IO/SMBus.cpp

```diff
@@ -0,0 +1,72 @@
+#include <fcntl.h>
+#include <sys/ioctl.h>
+#include <spdlog/spdlog.h>
+#include <unistd.h>
+#include "SMBus.h"
+
+extern "C" {
+#include <i2c/smbus.h>
+#include <linux/i2c-dev.h>
+}
```

Comment:
This is needed because the Linux I2C libraries are not quite C++ ready. If we don't have the `extern` block, we run into linker issues.